pkg/apis/kubeone/v1alpha1/types.go

```diff
@@ -180,6 +180,7 @@ type WorkerConfig struct {
 // ProviderSpec describes a worker node
 type ProviderSpec struct {
 	CloudProviderSpec   json.RawMessage   `json:"cloudProviderSpec"`
+	Annotations         map[string]string `json:"annotations,omitempty"`
```

Comment:
The v1alpha1 is frozen, so I don't think we should change it.

@kron4eg Any opinions on this?

Reply:
Yeah, let's not touch frozen stuff

Reply:
I included it for `v1alpha1` too, because of [this line](https://github.com/kubermatic/kubeone/blob/master/pkg/templates/machinecontroller/machines.go#L132). Here `v1alpha1` is used, so it's needed in the types.
Is it save to update this line to `v1beta1`?

Reply:
Those are different APIs. The API you've linked is Cluster-API v1alpha1 ([imported from machine-controller](https://github.com/kubermatic/kubeone/blob/ec0e690a684fbbbac0473c73d571ea207238ea96/pkg/templates/machinecontroller/machines.go#L32)), while this is the KubeOne's API.

Reply:
I included it for `v1alpha1` because of the conversions. Otherwise I get the following warning for `hack/update-codegen.sh`:
```
Generating deepcopy funcs
Generating defaulters
Generating conversions
E1125 15:24:08.526211    8987 conversion.go:755] Warning: could not find nor generate a final Conversion function for k8c.io/kubeone/pkg/apis/kubeone.ProviderSpec -> k8c.io/kubeone/pkg/apis/kubeone/v1alpha1.ProviderSpec
E1125 15:24:08.526277    8987 conversion.go:756]   the following fields need manual conversion:
E1125 15:24:08.526281    8987 conversion.go:758]       - Annotations
```
If this is fine, I can remove the `Annotations` field for `v1alpha`.

@xmudrii @kron4eg  What is your opinion about this?

Reply:
You guys are too fast.

Yeah, I realized I mixed APIs here and deleted my comment. Sorry for the mess.

Reply:
You can solve that by adding a custom conversion function that would call the autogenerated one. The autogenerated function doesn't have a conversion for the `Annotations` field, but we don't need it as that field doesn't exist in the `v1alpha1` API.

You can do it such as:

- Remove the Annotations field from the `v1alpha1` API
- Run the code generation script (`./hack/update-codegen.sh`)
- Add the following function to the [`pkg/apis/kubeone/v1alpha1/conversion.go` file](https://github.com/kubermatic/kubeone/blob/master/pkg/apis/kubeone/v1alpha1/conversion.go)
```go
func Convert_kubeone_ProviderSpec_To_v1alpha1_ProviderSpec(in *kubeoneapi.ProviderSpec, out *ProviderSpec, s conversion.Scope) error {
	if err := autoConvert_kubeone_ProviderSpec_To_v1alpha1_ProviderSpec(in, out, s); err != nil {
		return err
	}

	// The Annotations field is not available in the v1alpha1 API.

	return nil
}
```
- Re-run the code generation to ensure everything is okay

Reply:
Done.